CONTRIBUTING.md

```diff
@@ -1,16 +1,239 @@
-# CONTRIBUTE
-This is a short version of the [Development Procedures](https://github.com/RIOT-OS/RIOT/wiki/Development-procedures).
+# Contributing to RIOT
 
-1. Check if your code follows the [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions). If the code does not comply these style rules, your code will not be merged.
+Thank you for your interest in contributing to RIOT! There are many ways to
+contribute, and we appreciate all of them. You can jump to the major sections
+of this document using the following links:
 
-2. The master branch should always be in a working state. The RIOT maintainers will create release tags based on this branch, whenever a milestone is completed.
+* [Feature Requests](#feature-requests)
+* [Bug Reports](#bug-reports)
+* [Pull Requests](#pull-requests)
+* [Writing Documentation](#writing-documentation)
 
-3. Comments on a pull request should be added to the request itself, and *not* to the commit.
+If you have questions, please send an email to users@riot-os.org or
+devel@riot-os.org mailing list or chat on
+[#riot-os](http://webchat.freenode.net?channels=riot-os).
 
-4. Keep commits to the point, e.g., don't add whitespace/typo fixes to other code changes. If changes are layered, layer the patches.
+As a reminder, all contributors are expected to follow our
+[Code of Conduct](CODE_OF_CONDUCT.md).
 
-5. Describe the technical detail of the change(s) as specific as possible.
+## Feature Requests
+[feature-requests]: #feature-requests
 
-6. Use [Labels](https://github.com/RIOT-OS/RIOT/wiki/Labels) to help classify pull requests and issues.
+Before opening a new feature request, check the
+[existing feature requests](https://github.com/RIOT-OS/RIOT/issues?q=state:open+type:issue+label:"Feature+Request")
+if there's one already opened on the same topic.
 
-7. Please adhere to our [code of conduct](https://github.com/RIOT-OS/RIOT/blob/master/CODE_OF_CONDUCT.md)
+To request new features or enhancements, just open a new
+[feature request issue](https://github.com/RIOT-OS/RIOT/issues/new?template=feature_request.md).
+Describe your needs and use cases if any. Give detail on why this is important
+and/or necessary for RIOT.
+
+## Bug Reports
+[bug-reports]: #bug-reports
+
+While bugs are unfortunate, they're a reality in software. We can't fix what we
+don't know about, so please report liberally. If you're not sure if something
+is a bug or not, feel free to file a bug anyway.
+
+**If you believe reporting your bug publicly represents a security risk to
+RIOT users, please send an email describing the bug to security@riot-os.org**.
+
```

Comment:
Suggestion: "We would appreciate waiting until the next but one release to fix the bug before reporting it on public channels, to allow us adequate time to fix it"?
I think it's important for the industry that white hats report security holes in IoT software, so this comment would allude to working with them on this.

Reply:
To clarify: this sentence would come after rather than instead of the text above. I.e., add something to the effect that they should allow us time to fix before warning everyone that our software's unsecure.

Reply:
No problem, this is how I've understand it and actually changed it

Reply:
ah sorry, I didn't see that.